client/trino-client/src/main/java/io/trino/client/JsonResponse.java

```diff
@@ -140,6 +141,11 @@ public static <T> JsonResponse<T> execute(JsonCodec<T> codec, Call.Factory clien
                 }
                 return new JsonResponse<>(response.code(), response.headers(), body, value, exception);
             }
+
+            if (response.isRedirect() && !((OkHttpClient) client).followRedirects()) {
+                throw new ClientException("Client redirection was detected, but 'disableFollowRedirects' flag is set to true");
+            }
```

Comment:
I don't feed good about this downcasting. Also the check should be handled by upper layer, not inside JsonResponse.